admin/tabs/general/alertBanner/AlertBannerPanel.ts

```diff
@@ -188,3 +201,67 @@ const previewPanel = hoistCmp.factory<AlertBannerModel>(({model}) => {
         ]
     });
 });
+
+//------------------
+// Preset Menu
+//------------------
+const presetMenu = hoistCmp.factory<AlertBannerModel>({
+    render({model}) {
+        const savedPresets = model.savedPresets,
+            {message, intent, iconName, enableClose} = model.formModel.values,
+            items = [];
+
+        if (isEmpty(savedPresets)) {
+            items.push(menuItem({text: 'No presets saved...', disabled: true}));
+        } else {
+            items.push(...savedPresets.map(preset => presetMenuItem({preset})));
+        }
+
+        items.push(
+            menuDivider(),
+            menuItem({
+                icon: Icon.add({intent: 'success'}),
+                disabled:
+                    !model.formModel.fields.message.value ||
+                    model.isPreset({message, intent, iconName, enableClose}),
+                text: 'Add current',
+                onClick: e => {
+                    model.addPreset();
+                    e.stopPropagation();
+                }
+            })
+        );
+
+        return menu({items, style: {maxHeight: '400px', overflowY: 'auto'}});
+    }
+});
+
+const presetMenuItem = hoistCmp.factory<AlertBannerModel>({
+    render({model, preset}) {
+        const {iconName, intent, message, dateCreated} = preset;
+        return menuItem({
+            icon: iconName
+                ? Icon.icon({iconName, intent, prefix: 'fas', size: 'lg'})
+                : Icon.placeholder({size: 'lg'}),
+            text: vbox({
+                items: [
+                    truncate(message, {length: 50}),
+                    span({
+                        item: `Saved by: ${XH.getUser().username} (${dateCreated})`,
```

Comment:
This makes it look like the current user saved every preset, regardless of who actually did! 

You want to pass `createdBy: XH.getUsername()` along with the timestamp when saving a new preset.  That will capture the user who is saving that preset and persist it to the shared blob on the server, so everyone else will see that they were the ones who created it.